sdk/client/src/otel/index.ts

```diff
@@ -0,0 +1,319 @@
+import {
+	CompositePropagator,
+	W3CBaggagePropagator,
+	W3CTraceContextPropagator,
+} from '@opentelemetry/core'
+import {
+	BatchSpanProcessor,
+	BufferConfig,
+	ConsoleSpanExporter,
+	ReadableSpan,
+	SimpleSpanProcessor,
+	SpanExporter,
+	WebTracerProvider,
+} from '@opentelemetry/sdk-trace-web'
+import { DocumentLoadInstrumentation } from '@opentelemetry/instrumentation-document-load'
+import { FetchInstrumentation } from '@opentelemetry/instrumentation-fetch'
+import { registerInstrumentations } from '@opentelemetry/instrumentation'
+import { XMLHttpRequestInstrumentation } from '@opentelemetry/instrumentation-xml-http-request'
+import { Resource } from '@opentelemetry/resources'
+import { OTLPTraceExporter } from '@opentelemetry/exporter-trace-otlp-http'
+import {
+	SEMRESATTRS_DEPLOYMENT_ENVIRONMENT,
+	SEMRESATTRS_SERVICE_NAME,
+} from '@opentelemetry/semantic-conventions'
+import { BatchSpanProcessorBase } from '@opentelemetry/sdk-trace-base/build/src/export/BatchSpanProcessorBase'
+import * as api from '@opentelemetry/api'
+import {
+	BrowserXHR,
+	getBodyThatShouldBeRecorded,
+} from '../listeners/network-listener/utils/xhr-listener'
+import type { NetworkRecordingOptions } from '../types/client'
+import { sanitizeHeaders } from '../listeners/network-listener/utils/network-sanitizer'
+import { shouldNetworkRequestBeTraced } from '../listeners/network-listener/utils/utils'
+import { UserInteractionInstrumentation } from './user-interaction'
+
+export type OtelConfig = {
+	projectId: string | number
+	sessionSecureId: string
+	endpoint?: string
+	environment?: string
+	networkRecordingOptions?: NetworkRecordingOptions
+	serviceName?: string
+	tracingOrigins?: boolean | (string | RegExp)[]
+}
+
+let provider: WebTracerProvider
+
+export const initializeOtel = (config: OtelConfig) => {
+	if (provider !== undefined) {
+		console.warn('OTEL already initialized. Skipping...')
+		return
+	}
+
+	// TODO: Is there a better way to determine if we're in dev mode? Seems
+	// import.meta.env.DEV is set to dev when building with dev:frontend.
```

Comment:
do we need this check? why don't we just keep the fallback always as `https://otel.highlight.io:4318`, requiring something to be provided via `config.endpoint` otherwise?